at-dungnguyen/Final/main.rb

```diff
@@ -0,0 +1,279 @@
+  def notification_management
+    p 'Notification Management'
+    p '======================='
+    p '1. List Notification'
+    p '2. Read a notification'
+    p '3. Back'
+    p '4. Logout'
+    p '5. Send a notification' if @user['role'] == SessionController::LEADER
+    p '6. Delete a notification' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_notify(function)
+    case function
+    when 1
+      all_notify
+    when 2
+      g_notify
+    when 3
+      @flag_notify = false
+    when 4
+      @flag_notify = false
+      @user = nil
+    when 5
+      send_notify
+    when 6
+      del_notify
+    when 0
+      @flag_notify = false
+      execute2(0)
+    end
+  end
+
+  def all_notify
+    list_notify = @nc.get_items(@user['team'])
+    if list_notify.any?
+      list_notify.each do |notify|
+        puts "#{notify['title']} | #{notify['publish_date']}"
+      end
+    end
+    p '====================================================================='
+  end
+
+  def g_notify
+    p 'Input id notification'
+    id = input
+    notify = @nc.get_item(id)
+    p "#{notify['title']} | #{notify['publish_date']}"
+    p notify['content']
+  end
+
+  def send_notify
+    p 'Input title'
+    title = gets.chomp
+    p 'Input content'
+    content = gets.chomp
+    publish_date = Time.now
+    id_team = @user['team']
+    notify = Notification.new(title, content, publish_date.strftime('%F %T'), id_team)
+    if @nc.add_item(notify)
+      p 'Send notify success'
+    else
+      p 'Send error!'
+    end
+  end
+
+  def del_notify
+    p 'Input id notification'
+    id = input
+    if @nc.del_item(id)
+      p 'Delete Notification Success'
+    else
+      p "Notification doesn't exist!"
+    end
+  end
+
+  def leave_management
+    p 'Leave Management'
+    p '==============='
+    p '1. Send a leave request'
+    p '2. Update a leave request'
+    p '3. Show list of leaves of employees.'
+    p '4. Back'
+    p '5. Logout'
+    p '6. Show list leaves request need approve' if @user['role'] == SessionController::LEADER
+    p '7. Approve or reject leave request' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  # Default
+  def screen1
+    p 'Welcome to Human Resources!'
+    p '============================'
+    p '1. Log in'
+    p '2. Register'
+    p '3. Forgot password'
+    p '0. Exit'
+    p '============================'
+    p 'Please choose a function'
+  end
+
+  def screen2
+    p 'Welcome to Human Resources!'
+    p '============================'
+    p '1. Team Management'
+    p '2. Leave management'
+    p '3. Notification management'
+    p '4. Logout'
+    p '5. CSV' if @user['role'] == SessionController::ADMIN
+    p '0. Exit'
+    p '============================'
+  end
+
+  def run
+    @ch = true
+    while @ch
+      screen1
+      function = input
+      clear
+      execute(function) if (0..3).cover? function
+    end
+  end
+
+  def execute2(function)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/59f7885620a5f3c551417f8e71806b1b95622bf8/at-dungnguyen/Final/main.rb#L237"></span>Metrics/CyclomaticComplexity: Cyclomatic complexity for execute2 is too high. [12/10]</td>
    </tr>
  </tbody>
</table>